### PR TITLE
Do not test with SecurityManager when Java version 24 and above

### LINKED ISSUE #1779: Ant Test CoverageTaskTest.testSecurityManager fails since JEP 486
Since integration of [JEP 486](https://openjdk.org/jeps/486) in https://github.com/openjdk/jdk/commit/db85090553ab14a84c3ed0a2604dd56c5b6e6982 the Ant test CoverageTaskTest.testSecurityManager fails:

    testSecurityManager:
    [jacoco:coverage] Enhancing java with coverage
         [java] Error occurred during initialization of VM
         [java] java.lang.Error: A command line option has attempted to allow or enable the Security Manager. Enabling a Security Manager is not supported.
         [java] 	at java.lang.System.initPhase3(java.base@24-internal/System.java:2071)
         [java] 

I see two options to address this:

1. Remove this test
2. Execute this test only for JDK < 24, for example with JUnit assumptions

### PR DESCRIPTION
**Pull request by marchof**
Fixes #1779

### CODE REVIEW

File: org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2024 Mountainminds GmbH & Co. KG and Contributors
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Brock Janiczak - initial API and implementation
```
Comment (Godin) [inline org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java:10]:
```suggestion
 *    Marc R. Hoffmann - initial API and implementation
```

File: org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<!--
+   Copyright (c) 2009, 2024 Mountainminds GmbH & Co. KG and Contributors
+   This program and the accompanying materials are made available under
+   the terms of the Eclipse Public License 2.0 which is available at
+   http://www.eclipse.org/legal/epl-2.0
+
+   SPDX-License-Identifier: EPL-2.0
+
+   Contributors:
+      Brock Janiczak - initial API and implementation
```
Comment (Godin) [inline org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.xml:12]:
```suggestion
      Marc R. Hoffmann - initial API and implementation
```

File: org.jacoco.ant.test/src/org/jacoco/ant/empty.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<!--
+   Copyright (c) 2009, 2024 Mountainminds GmbH & Co. KG and Contributors
+   This program and the accompanying materials are made available under
+   the terms of the Eclipse Public License 2.0 which is available at
+   http://www.eclipse.org/legal/epl-2.0
+
+   SPDX-License-Identifier: EPL-2.0
+
+   Contributors:
+      Brock Janiczak - initial API and implementation
```
Comment (Godin) [inline org.jacoco.ant.test/src/org/jacoco/ant/empty.xml:12]:
```suggestion
      Marc R. Hoffmann - initial API and implementation
```

File: org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	public static TestSuite suite() {
+		if (JavaVersion.current().isBefore("24")) {
+			System.setProperty("org.jacoco.ant.securityManagerTest.classes.dir",
+					TestTarget.getClassPath());
+			final File file = new File(
+					"src/org/jacoco/ant/SecurityManagerTest.xml");
+			return new AntUnitSuite(file, SecurityManagerTest.class);
+		}
+		// No SecurityManager any more in Java 24
+		final File file = new File("src/org/jacoco/ant/empty.xml");
```
Comment (Godin) [inline org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java:36]: @marchof I guess that `AntUnitSuiteRunner` does not understand `org.junit.AssumptionViolatedException` that we use in other places and that's why we need `empty.xml` here?

**Review by Godin — CHANGES_REQUESTED** [review]

File: org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java (hunk 4 above)
Comment (marchof) [reply org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java:36]: No, this results in an initialization exception. Also I tried to return an empty `TestSuite` but the `AntUnitSuiteRunner` expects a `AntUnitSuite` instance.

File: org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+	public static TestSuite suite() {
+		if (JavaVersion.current().isBefore("24")) {
+			System.setProperty("org.jacoco.ant.securityManagerTest.classes.dir",
+					TestTarget.getClassPath());
+			final File file = new File(
+					"src/org/jacoco/ant/SecurityManagerTest.xml");
+			return new AntUnitSuite(file, SecurityManagerTest.class);
+		}
+		// No SecurityManager any more in Java 24
```
Comment (Godin) [inline org.jacoco.ant.test/src/org/jacoco/ant/SecurityManagerTest.java:35]:
```suggestion
		// Ability to enable Security Manager was removed in Java 24
		// https://openjdk.org/jeps/486
```

**Review by Godin — APPROVED** [review]

**Comment by Godin** [comment]
@marchof thanks!
FYI Azure Pipelines now use JDK 24 EA b24 that contains integration of JEP 486

**Pull request merged**